CONTRIBUTING.md

```diff
@@ -1,9 +1,197 @@
+of `nemos` in your GitHub account. You should then clone *your fork* to your local machine.
+
+```bash
+git clone https://github.com/<YourUserName>/nemos.git
+cd nemos
+```
+
+2) Install `nemos` in editable mode with developer dependencies  
+
+```bash
+pip install -e .[dev]
+```
+
+> **_NOTE:_**  In order to install `nemos` in editable mode you will need a Python virtual environment. Please see our documentation [here](https://nemos.readthedocs.io/en/latest/installation/) that 
+> provides guidance on how to create and activate a virtual environment.
+
+3) Add the upstream branch:
+
+```bash
+git remote add upstream https://github.com/flatironinstitute/nemos
+```
+
+At this point you have two remotes: `origin` (your fork) and `upstream` (the canonical version). You won't have permission to push to upstream (only `origin`), but 
+this make it easy to keep your `nemos` up-to-date with the canonical version by pulling from upstream: `git pull upstream`.
+
+#### Creating a new branch
+
+As mentioned previously, each feature in `nemos` is worked on in a separate branch. This allows multiple people developing multiple features simultaneously, without interfering with each other's work. To create
+your own branch, run the following from within your `nemos` directory:
+
+> **_NOTE:_** Below we are checking out the `development` branch. In terms of the `nemos` contribution workflow cycle, the `development` branch accumulates a series of
+> changes from different feature branches that are then all merged into the `main` branch at one time (normally at the time of a release) :D
+
+```bash
+# switch to the development branch
+git checkout development
+# update your development branch
+git pull origin development
+# sync with upstream development
+git pull upstream development
+# update your fork's development branch with any changs from upstream
+git push origin development
+# create and switch to a new branch
+git checkout -b my_feature_branch
+```
+
+After you have made changes on this branch, add and commit them when you are ready:
+
+```bash
+# stage the changes
+git add src/nemos/the_changed_file.py
+# commit your changes
+git commit -m "A one-line message explaining the changes made"
+# push to the remote origin
+git push origin my_feature_branch
+```
+
+#### Contributing your change back to NeMoS
+
+You can make any number of changes on your branch. Once you are happy with your changes, add tests to check that they run correctly and add documentation to properly note your changes. 
+See below for details on how to [add tests](#adding-tests) and properly [document](#adding-documentation) your code.
+
+Lastly, you should make sure that the existing tests all run successfully and that the codebase is formatted properly:
+
+```bash
+# run tests and make sure they all pass
+pytest tests/
+# format the code base
+black src/
+isort src
+flake8 --config=tox.ini src
+```
```

Comment:
is it worth telling people to just use tox here? that way they only need to run one command (and we know it's the same as the tests)